Code/EnginePlugins/RmlUiPlugin/Implementation/EventListener.cpp

```diff
@@ -0,0 +1,57 @@
+#include <RmlUiPluginPCH.h>
+
+#include <RmlUiPlugin/Implementation/EventListener.h>
+#include <RmlUiPlugin/RmlUiContext.h>
+
+namespace ezRmlUiInternal
+{
+  static EventListenerInstancer* s_pInstancer;
+
+  void EventListener::ProcessEvent(Rml::Core::Event& event)
+  {
+    static_cast<ezRmlUiContext*>(event.GetTargetElement()->GetContext())->ProcessEvent(m_sIdentifier, event);
+  }
+
+  void EventListener::OnDetach(Rml::Core::Element* element)
+  {
+    s_pInstancer->ReturnToPool(*this);
+  }
+
+  //////////////////////////////////////////////////////////////////////////
+
+  EventListenerInstancer::EventListenerInstancer()
+  {
+    s_pInstancer = this;
+  }
+
+  EventListenerInstancer::~EventListenerInstancer()
+  {
+    s_pInstancer = nullptr;
+  }
+
+  Rml::Core::EventListener* EventListenerInstancer::InstanceEventListener(const Rml::Core::String& value, Rml::Core::Element* element)
```

Comment:
"Instantiate" instead of "Instance" ?

Reply:
This is an interface implementation, I can't change the name